lib/api/controllers/adminController.js

```diff
@@ -42,10 +41,15 @@ class AdminController {
   /**
    * Reset the internal storage components (internalEngine index, cache and memory storage)
    */
-  resetKuzzleData () {
-    const options = { refresh: 'wait_for' };
+  resetKuzzleData (request) {
+    if (_locks.resetKuzzleData) {
+      throw new BadRequestError('Kuzzle is already reseting his data.');
```

Comment:
1. `resetting`
2. Should not be a `BadRequestError`: the request is correct. I suggest throwing a [PreconditionError](https://docs.kuzzle.io/api-documentation/errors#preconditionerror) instead